packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/diagram/StylesFactory.java

```diff
@@ -76,73 +78,44 @@ public EdgeStyle createEdgeStyle(org.eclipse.sirius.components.view.EdgeStyle ed
-                                             .build();
-                // @formatter:on
-            }
-            break;
-        case NodeType.NODE_LIST:
+        if (nodeStyle instanceof RectangularNodeStyleDescription) {
             // @formatter:off
-            result = ListNodeStyle.newListNodeStyle()
-                                  .color(Optional.ofNullable(nodeStyle.getColor()).orElse(DEFAULT_COLOR))
-                                  .borderColor(Optional.ofNullable(nodeStyle.getBorderColor()).orElse(DEFAULT_COLOR))
-                                  .borderSize(nodeStyle.getBorderSize())
-                                  .borderStyle(LineStyle.valueOf(nodeStyle.getBorderLineStyle().getLiteral()))
-                                  .borderRadius(nodeStyle.getBorderRadius())
-                                  .build();
+            result = RectangularNodeStyle.newRectangularNodeStyle()
+                   .withHeader(((RectangularNodeStyleDescription) nodeStyle).isWithHeader())
+                   .color(Optional.ofNullable(nodeStyle.getColor()).orElse(DEFAULT_COLOR))
+                   .borderColor(Optional.ofNullable(nodeStyle.getBorderColor()).orElse(DEFAULT_COLOR))
+                   .borderSize(nodeStyle.getBorderSize())
+                   .borderStyle(LineStyle.valueOf(nodeStyle.getBorderLineStyle().getLiteral()))
+                   .borderRadius(nodeStyle.getBorderRadius())
+                   .build();
             // @formatter:on
-            break;
-        case NodeType.NODE_LIST_ITEM:
-            result = ListItemNodeStyle.newListItemNodeStyle().backgroundColor("transparent").build(); //$NON-NLS-1$
-            break;
-        case NodeType.NODE_RECTANGLE:
-        default:
+        } else if (nodeStyle instanceof IconLabelNodeStyleDescription) {
+            result = IconLabelNodeStyle.newIconLabelNodeStyle().backgroundColor("transparent").build(); //$NON-NLS-1$
+        } else if (nodeStyle instanceof ImageNodeStyleDescription) {
             // @formatter:off
-            result = RectangularNodeStyle.newRectangularNodeStyle()
-                                         .color(Optional.ofNullable(nodeStyle.getColor()).orElse(DEFAULT_COLOR))
-                                         .borderColor(Optional.ofNullable(nodeStyle.getBorderColor()).orElse(DEFAULT_COLOR))
-                                         .borderSize(nodeStyle.getBorderSize())
-                                         .borderStyle(LineStyle.valueOf(nodeStyle.getBorderLineStyle().getLiteral()))
-                                         .borderRadius(nodeStyle.getBorderRadius())
-                                         .build();
+            result = ImageNodeStyle.newImageNodeStyle()
+                    .scalingFactor(1)
+                    .imageURL("/custom/" + optionalEditingContextId.get().toString() + "/" + ((ImageNodeStyleDescription) nodeStyle).getShape()) //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
`Optional.get()` called on `optionalEditingContextId` without checking first.

Reply:
I do not have this version anymore